### PR TITLE
Base Price Storage model fix.

### PR DESCRIPTION
**Pull request by wassuupp**
Fix V1/products/base-prices endpoint to update prices for Catalog Price Mode - Website

### Description (*)
V1/products/base-prices endpoint does not work with Catalog Price Mode - Website
Function was added to affect prices for all store views inside changed website

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by wassuupp** [comment]
@magento run all tests

File: app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php [hunk 1]
```diff
@@ -146,6 +177,16 @@ public function update(array $prices)
             }
         }
 
+        try {
```
Comment (nuzil) [inline app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php:180]: I would not add here try / catch. Cause if such attribute do not exists, exception will be much earlier, already on $this->getPricePersistence() method

File: app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php [hunk 2]
```diff
@@ -146,6 +177,16 @@ public function update(array $prices)
...
+            $priceAttribute = $this->productAttributeRepository->get($this->attributeCode);
+        } catch (NoSuchEntityException $e) {
+            $priceAttribute = null;
+        }
+
+        if ($priceAttribute !== null && $priceAttribute->isScopeWebsite()) {
+            $formattedPrices = $this->applyWebsitePrices($formattedPrices);
```
Comment (nuzil) [inline app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php:187]: So lets put this together with previous part

File: app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php [hunk 3]
```diff
@@ -225,4 +266,35 @@ private function retrieveValidPrices(array $prices)
...
+    {
+        foreach ($formattedPrices as $price) {
+            if ($price['store_id'] == Store::DEFAULT_STORE_ID) {
+                continue;
+            }
+
+            try {
+                $storeIds = $this->storeManager->getStore($price['store_id'])->getWebsite()->getStoreIds();
+            } catch (NoSuchEntityException $e) {
+                continue;
```
Comment (nuzil) [inline app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php:286]: I would throw here an exception directly without allowing user to even save any prices

**Review by nuzil — CHANGES_REQUESTED** [review]
Can you also cover this fix with tests?

**Comment by VladimirZaets** [comment]
@wassuupp , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Comment by nuzil** [comment]
@VladimirZaets  ... wait a sec, I will push him :-D

**Pull request closed without merging**